src/vm/eventpipe.cpp

```diff
@@ -732,6 +711,50 @@ StackWalkAction EventPipe::StackWalkCallback(CrawlFrame *pCf, StackContents *pDa
     return SWA_CONTINUE;
 }
 
+uint32_t EventPipe::GenerateSessionIndex()
+{
+    LIMITED_METHOD_CONTRACT;
+    PRECONDITION(IsLockOwnedByCurrentThread());
+
+    uint64_t id = 1;
+    for (uint32_t i = 0; i < MaxNumberOfSessions; ++i, id <<= i)
```

Comment:
The `id <<= i` is not correct - it would have generated a sequence of id to be 1,2,8,64,...